src/store/data-explorer/data-explorer-action.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) The Arvados Authors. All rights reserved.
+//
+// SPDX-License-Identifier: AGPL-3.0
+
+import { default as unionize, ofType, UnionOf } from "unionize";
+import { SortDirection, DataColumn } from "../../components/data-table/data-column";
+import { DataTableFilterItem } from "../../components/data-table-filters/data-table-filters";
+import { DataColumns } from "../../components/data-table/data-table";
+
+type WithId<T> = T & { id: string };
```

Comment:
not sure about it.. it doesn't even significantly reduce code size. I would prefer inlining such things.